modules/lightmapper_rd/lm_compute.glsl

```diff
@@ -735,4 +749,118 @@ void main() {
 	imageStore(dest_light, ivec3(atlas_pos, params.atlas_slice), c);
 
 #endif
+
+#ifdef MODE_DENOISE
+	// Joint Non-local means (JNLM) denoiser.
+	// Based on YoctoImageDenoiser's JNLM implementation with corrections from "Nonlinearly Weighted First-order Regression for Denoising Monte Carlo Renderings".
+	// <https://github.com/ManuelPrandini/YoctoImageDenoiser/blob/06e19489dd64e47792acffde536393802ba48607/libs/yocto_extension/yocto_extension.cpp#L207>
+	// <https://benedikt-bitterli.me/nfor/nfor.pdf>
```

Comment:
We should include the full copyright notice. While its kind of awkward here, I think this chunk of code likely counts as a "substantial portion". We would also need to include the copyright info in thirdparty I beleive CC @akien-mga 
```
// LICENSE:
//
// Copyright (c) 2020 -- 2020 Fabio Pellacini
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.
```

Reply:
For the copyright info, you'd add it like this:
```diff
diff --git a/COPYRIGHT.txt b/COPYRIGHT.txt
index 500eb84d08..9abf526209 100644
--- a/COPYRIGHT.txt
+++ b/COPYRIGHT.txt
@@ -63,6 +63,11 @@ Copyright: 2011, Ole Kniemeyer, MAXON, www.maxon.net
  2007-2014, Juan Linietsky, Ariel Manzur
 License: Expat and Zlib
 
+Files: ./modules/lightmapper_rd/lm_compute.glsl
+Comment: Joint Non-Local Means (JNLM) denoiser
+Copyright: 2020, Fabio Pellacini
+License: Expat
+
 Files: ./platform/android/java/lib/aidl/com/android/*
  ./platform/android/java/lib/res/layout/status_bar_ongoing_event_progress_bar.xml
  ./platform/android/java/lib/src/com/google/android/*
```

If the code was substantially modified for Godot, then you could also add the typical Godot copyright:
```diffdiff --git a/COPYRIGHT.txt b/COPYRIGHT.txt
index 500eb84d08..7edda06c1b 100644
--- a/COPYRIGHT.txt
+++ b/COPYRIGHT.txt
@@ -63,6 +63,13 @@ Copyright: 2011, Ole Kniemeyer, MAXON, www.maxon.net
  2007-2014, Juan Linietsky, Ariel Manzur
 License: Expat and Zlib
 
+Files: ./modules/lightmapper_rd/lm_compute.glsl
+Comment: Joint Non-Local Means (JNLM) denoiser
+Copyright: 2020, Fabio Pellacini
+ 2014-present, Godot Engine contributors
+ 2007-2014, Juan Linietsky, Ariel Manzur
+License: Expat
+
 Files: ./platform/android/java/lib/aidl/com/android/*
  ./platform/android/java/lib/res/layout/status_bar_ongoing_event_progress_bar.xml
  ./platform/android/java/lib/src/com/google/android/*
```

Reply:
It was definitely substantially modified. The license in the original repo is MIT. Do we have to specify expat or MIT?

Would we be fine with just adding this copyright notice or do we have to include the license in the code like Clay mentioned?

Reply:
The `COPYRIGHT.txt` file follows a Debian format and they use "Expat" for MIT, as they deem the MIT name too ambiguous (there's multiple versions of it with slightly different language). So it should use `Expat`.

And yeah I think it's good to include the full license in that function for good measure.

Reply:
Done.